drivers/platform/maxim/max32670/maxim_irq.c

```diff
@@ -0,0 +1,581 @@
+ *    patent holders to use this software.
+ *  - Use of the software either in source or binary form, must be run
+ *    on or directly connected to an Analog Devices Inc. component.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+/******************************************************************************/
+/************************* Include Files **************************************/
+/******************************************************************************/
+
+#include <stdint.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include <errno.h>
+#include "rtc.h"
+#include "uart.h"
+#include "tmr.h"
+#include "maxim_irq.h"
+#include "maxim_rtc.h"
+#include "max32670.h"
+#include "no_os_irq.h"
+#include "no_os_list.h"
+#include "no_os_uart.h"
+#include "no_os_rtc.h"
+#include "no_os_util.h"
+
+static struct event_list _events[] = {
+	[NO_OS_EVT_GPIO] = {.event = NO_OS_EVT_GPIO},
+	[NO_OS_EVT_UART_TX_COMPLETE] = {.event = NO_OS_EVT_UART_TX_COMPLETE},
+	[NO_OS_EVT_UART_RX_COMPLETE] = {.event = NO_OS_EVT_UART_RX_COMPLETE},
+	[NO_OS_EVT_UART_ERROR] = {.event = NO_OS_EVT_UART_ERROR},
+	[NO_OS_EVT_RTC] = {.event = NO_OS_EVT_RTC},
+	[NO_OS_EVT_TIM_ELAPSED] = {.event = NO_OS_EVT_TIM_ELAPSED},
+};
+
+extern mxc_uart_req_t uart_irq_state[MXC_UART_INSTANCES];
+extern bool is_callback;
+
+/******************************************************************************/
+/************************ Functions Definitions *******************************/
+/******************************************************************************/
+
+/**
+ * @brief Action comparator function
+ * @param data1 - List element
+ * @param data2 - Key
+ * @return 0 if the two are equal, any other integer otherwise
+ */
+int32_t irq_action_cmp(void *data1, void *data2)
+{
+	return ((struct irq_action *)data1)->irq_id -
+	       ((struct irq_action *)data2)->irq_id;
+}
+
+/**
+ * @brief Check if the irq_id is for a GPIO interrupt
+ * @param irq_id - The interrupt vector entry id.
+ * @return true/false
+ */
+static bool is_gpio_irq_id(uint32_t irq_id)
+{
+	for (uint32_t i = 0; i < MXC_CFG_GPIO_INSTANCES; i++) {
```

Comment:
nit: to allign better with the linux drivers code style, variables are declared at the start of the function. check the entire code for this occurrences.

Reply:
Will move variable declaration outside loop